sklearn/preprocessing/imputation.py

```diff
@@ -375,3 +375,173 @@ def transform(self, X):
+        If "auto", the transformed ``X`` will be os same type as input
+
+    Attributes
+    ----------
+    feat_with_missing_  : array of shape(n_missing_features, )
+        The features with missing values.
+        Note that this is only stored if features == 'train
+
+    n_features_ : int
+        The number of features during fit time.
+
+    Example
+    -------
+    >>> from sklearn.preprocessing import MissingIndicator
+    >>> import numpy as np
+    >>> a = MissingIndicator(missing_values = -1)
+    >>> X1 = np.array([
+    ...       [-1,  1,  3],
+    ...       [ 4,  0, -1],
+    ...       [ 8,  1,  0]
+    ... ])
+    >>> X2 = np.array([
+    ...       [ 5, -1, -1],
+    ...       [-1,  2,  3],
+    ...       [ 2,  4,  0]
+    ... ])
+    >>> MI = MissingIndicator(missing_values = -1)
+    >>> MI.fit(X1)
+    MissingIndicator(features='train', missing_values=-1, sparse='auto')
+    >>> X2_tr = MI.transform(X2)
+    >>> X2_tr
+    array([[0, 1],
+           [1, 0],
+           [0, 0]])
+
+    """
+
+    def __init__(self, missing_values="NaN", features="train", sparse="auto"):
+        self.missing_values = missing_values
+        self.features = features
+        self.sparse = sparse
+
+    def fit(self, X, y=None):
+        """Fit the transformer on X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape (n_samples, n_features)
+            Input data, where ``n_samples`` is the number of samples and
+            ``n_features`` is the number of features.
+
+        Returns
+        -------
+        self : object
+            Returns self.
+        """
+        if (isinstance(self.features, six.string_types) and
+                self.features not in ["train", "all"]):
+            raise ValueError("Can only use these options: 'train', 'all'"
+                             " got {0}".format(self.features))
+        elif (isinstance(self.features, np.ndarray) and
+              not np.issubdtype(self.features.dtype, np.integer)):
+            raise ValueError("Features should be an array of integers")
+
+        if not ((isinstance(self.sparse, six.string_types) and
+                self.sparse == "auto") or isinstance(self.sparse, bool)):
+            raise ValueError("sparse can only use be boolean or 'auto'"
+                             " got {0}".format(self.sparse))
+
+        X = check_array(X, accept_sparse=('csc', 'csr'), dtype=np.float64)
+        self.n_features_ = X.shape[1]
+
+        if self.features == "train":
+            _, self.feat_with_missing_ = self._get_missing_features_info(X)
+
+        return self
+
+    def transform(self, X):
+        """Generate missing values indicator for X.
+
+        Parameters
+        ----------
+        X : {array-like, sparse matrix}, shape = [n_samples, n_features]
+            The input data to complete.
+
+        Returns
+        -------
+        Xt : {array-like, sparse matrix}, shape = [n_samples, n_features]
```

Comment:
"array or sparse matrix".

Cannot be array-like, only array, while "or" is clearer than set notation.